sw/device/tests/hmac/dif_hmac_test.c

```diff
@@ -0,0 +1,189 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_hmac.h"
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/common.h"
+#include "sw/device/lib/flash_ctrl.h"
+#include "sw/device/lib/uart.h"
+
+#define HMAC0_BASE_ADDR 0x40120000
+#define MAX_FIFO_FILL 10
+
+static const char kData[142] =
+    "Every one suspects himself of at least one of "
+    "the cardinal virtues, and this is mine: I am "
+    "one of the few honest people that I have ever "
+    "known";
+
+static const uint32_t kHmacKey[8] = {0x243f6a88, 0x85a308d3, 0x13198a2e,
+                                     0x03707344, 0xa4093822, 0x299f31d0,
+                                     0x082efa98, 0xec4e6c89};
+
+static const uint32_t kExpectedShaDigest[8] = {
+    0xdc96c23d, 0xaf36e268, 0xcb68ff71, 0xe92f76e2,
+    0xb8a8379d, 0x426dc745, 0x19f5cff7, 0x4ec9c6d6};
+
+static const uint32_t kExpectedHmacDigest[8] = {
+    0xe4987b39, 0x3f83d390, 0xc2f3bbaf, 0x3195dbfa,
+    0x23fb480c, 0xb012ae5e, 0xf1394d28, 0x1940ceeb};
+
+static void print_uint(uint64_t n) {
```

Comment:
`uart_send_uint` sends hex, which is fine for some things, but a bit cumbersome when you're just looking for lengths of data sent or an array index. `uart_send_uint` also prints a fixed length string by `size`, so it would yield error strings in the form 

> Digest mismatch, expected: 0x01234567 at index [00000001] but got: 0x89ABCDEF.

> Message length mismatch. Expected 000000AB bits but got 000000CD bits.

I just think it's cleaner to have decimal outputs with no leading zeros. Happy to move this to uart.[ch] if it's helpful.

Reply:
Oh, actually maybe it's worth using @mcy recently landed `print_log`. It is capable of printing ints and hex, not sure about 64bit though.